### ISSUE TITLE
Error while passing json output as string

### ISSUE THREAD

**Issue by adurai81** [helm/helm#6854]
Output of `helm version`:
```
Client: &version.Version{SemVer:"v2.14.3", GitCommit:"0e7f3b6637f7af8fcfddb3d2941fcc7cbebb0085", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.14.3", GitCommit:"0e7f3b6637f7af8fcfddb3d2941fcc7cbebb0085", GitTreeState:"clean"}
```

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.3", GitCommit:"2d3c76f9091b6bec110a5e63777c332469e0cba2", GitTreeState:"clean", BuildDate:"2019-08-19T12:38:36Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.5-6+d6ea2e3ed7815b", GitCommit:"d6ea2e3ed7815b9b53d854038041f43b0a98555e", GitTreeState:"clean", BuildDate:"2019-09-19T23:10:35Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): Oracle Cloud Infra

Problem Statement:
I'm trying to pass JWKS string (JSON) as value to a parameter that would eventually be replaced in the yaml file.

This is the most minimal string for reproducing the issue -
```
helm install .  --debug --dry-run --name test --namespace istio-system --set authN.jwks='{"keys":[{"kty":"RSA"}]}'
```

Error -
```
[debug] Created tunnel using local port: '60875'

[debug] SERVER: "127.0.0.1:60875"

[debug] Original chart version: ""
[debug] CHART PATH: LocalDirectoryListing

Error: failed parsing --set data: key "]}" has no value
```

To further clarify if its a simple string (something like `TestString`), it works.

**Comment by bacongobbler** [comment]
Does this work if you use `--set-string` instead?

**Comment by adurai81** [comment]
Unfortunately it doesn't work -

```
helm install .  --debug --dry-run --name test --namespace istio-system --set-string authN.jwks='{"keys":[{"kty":"RSA"}]}'

```

```
Error: failed parsing --set-string data: key "]}" has no value
```

**Comment by adurai81** [comment]
Any thoughts on this ? Is this a open issue ?

**Comment by bacongobbler** [comment]
at first glance, this appears to be a bug, yes.

Let us know if you find anything in the code base that would indicate where it may be failing.

**Comment by ziflex** [comment]
Same with Helm3.
```
version.BuildInfo{Version:"v3.0.2", GitCommit:"19e47ee3283ae98139d98460de796c1be1e3975f", GitTreeState:"clean", GoVersion:"go1.13.5"}
```
I'm using RabbitMQ Chart and need to pass definitions via command line. 
Neither ``--set`` nor ``--set-string`` works. 
It seems that Helm tries to parse a given string and create Go value (slice or struct).

**Comment by andrewmcguinness** [comment]
I have seen a similar error with
`--set-string variable="{astringwith}andsoon"
`
which results in values that look like

```
variable:
  - astringwith
```
instead of
```
variable: '{astringwith}andsoon'
```
This would be reasonable for --set, but isn't --set-string supposed to force the value to be treated as a string?

**Comment by bacongobbler** [comment]
No. `--set-string` coerces the value into a string, but it still retains the same syntax sugar as `--set`. See #4030.

**Comment by bacongobbler** [comment]
actually, it looks like this can be closed as a duplicate of #4030. Not sure why I didn't see that sooner.

Please follow up in that ticket.
